src/BasicRum/DiagramSchema.php

```diff
@@ -0,0 +1,391 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\BasicRum;
+
+use App\BasicRum\BusinessMetrics\Collaborator as BusinessMetrics;
+use App\BasicRum\DiagramSchema\Filters\Collaborator as Filters;
+use App\BasicRum\TechnicalMetrics\Collaborator as TechnicalMetrics;
+
+class DiagramSchema
+{
+    private $definitionSegment;
+    private $layout;
+    private $filters;
+    private $businessMetrics;
+    private $technicalMetrics;
+
+    /**
+     * DiagramSchema constructor.
+     *
+     * @throws \Exception
+     */
+    public function __construct($type)
+    {
+        $this->type = $type;
+        $this->generateLayout();
+
+        $this->generateGlobalFilters();
+        $this->generateDefinitionSegment();
+    }
+
+    public function getBusinessMetrics(): array
+    {
+        $class = new BusinessMetrics();
+        $businessMetricsClassMap = $class->getAllPossibleRequirements();
+
+        $count = \count($businessMetricsClassMap);
+
+        $segmentMetricsPart = [
+            'business_metrics' => [
+                'type' => 'object',
+                'properties' => [],
+            ],
+        ];
+
+        foreach (array_keys($businessMetricsClassMap) as $key) {
+            $segmentMetricsPart['business_metrics']['properties'][$key] = [
+                'type' => 'object',
+                'properties' => [],
+            ];
+
+            $segmentMetricsPart['business_metrics']['properties'][$key]['properties'] = $this->getDataFlavor($this->type);
+
+            $oneOf['oneOf'][] = [
+                'required' => [$key],
+            ];
+        }
+
+        $segmentMetricsPart['business_metrics'] = array_merge(
+            $segmentMetricsPart['business_metrics'],
+            $oneOf
+        );
+
+        return $segmentMetricsPart;
+    }
+
+    public function getTechnicalMetrics(): array
+    {
+        $class = new TechnicalMetrics();
+        $technicalMetricsClassMap = $class->getAllPossibleRequirements();
+
+        $segmentMetricsPart = [
+            'technical_metrics' => [
+                'type' => 'object',
+                'properties' => [],
+            ],
+        ];
+
+        foreach (array_keys($technicalMetricsClassMap) as $key) {
+            $segmentMetricsPart['technical_metrics']['properties'][$key] = [
+                'type' => 'object',
+                'properties' => [],
+            ];
+
+            $segmentMetricsPart['technical_metrics']['properties'][$key]['properties'] = $this->getDataFlavor($this->type);
+
+            $oneOf['oneOf'][] = [
+                'required' => [$key],
+            ];
+        }
+
+        $segmentMetricsPart['technical_metrics'] = array_merge(
+            $segmentMetricsPart['technical_metrics'],
+            $oneOf
+        );
+
+        return $segmentMetricsPart;
+    }
+
+    public function getDataFlavor($renderType): array
+    {
+        $dataFlavor = '';
```

Comment:
I think that we need to define an array here like:
```
$dataFlavor = [];
```

However, I have another recommendation about the whole function:

    public function getDataFlavor($renderType): array
    {
        $dataFlavor = '';
 

 
I think that we need to define an array here like:
```
$dataFlavor = [];
```

However, I have another recommendation about the whole function:

```
        if ('time_series' == $renderType) {
            return [
                'data_flavor' => [
                    'type' => 'object',
                    'properties' => [
                        'percentile' => [
                            'type' => 'integer',
                            'minimum' => 0,
                            'maximum' => 100,
                        ],
                    ],
                    'oneOf' => [
                        [
                            'required' => ['percentile'],
                        ],
                        [
                            'required' => ['histogram'],
                        ],
                        [
                            'required' => ['count'],
                        ],
                    ],
                ],
            ];
        }

        if ('plane' == $renderType) {
            return [
                'data_flavor' => [
                    'type' => 'object',
                    'properties' => [
                        'histogram' => [
                            'type' => 'object',
                            'properties' => [
                                'bucket' => [
                                    'enum' => [200],
                                    'type' => 'integer',
                                ],
                            ],
                        ],
                    ],
                    'oneOf' => [
                        [
                            'required' => ['percentile'],
                        ],
                        [
                            'required' => ['histogram'],
                        ],
                        [
                            'required' => ['count'],
                        ],
                    ],
                ],
            ];
        }
        
        if ('distribution' == $renderType) {
            return [
                'data_flavor' => [
                    'type' => 'object',
                    'properties' => [
                        'count' => [
                            'type' => 'boolean',
                        ],
                    ],
                    'oneOf' => [
                        [
                            'required' => ['percentile'],
                        ],
                        [
                            'required' => ['histogram'],
                        ],
                        [
                            'required' => ['count'],
                        ],
                    ],
                ],
            ];
        }

        throw new Exception('Missing render type.');
    }
```